finality-aleph/src/party/mod.rs

```diff
@@ -174,15 +174,44 @@ where
         run_justification_handler(handler, &spawn_handle.clone().into(), justification_rx);
 
     // Prepare and start the network
-    let network =
-        ConsensusNetwork::<NetworkData<B>, _, _>::new(network.clone(), "/cardinals/aleph/1".into());
-    let session_manager = network.session_manager();
+    let (commands_for_network, commands_from_io) = mpsc::unbounded();
+    let (messages_for_network, messages_from_user) = mpsc::unbounded();
+    let (commands_for_service, commands_from_user) = mpsc::unbounded();
+    let (messages_for_service, commands_from_manager) = mpsc::unbounded();
+    let (messages_for_user, messages_from_network) = mpsc::unbounded();
+
+    let connection_io = ConnectionIO::new(
+        commands_for_network,
+        messages_for_network,
+        commands_from_user,
+        commands_from_manager,
+        messages_from_network,
+    );
+    let connection_manager = ConnectionManager::<_, _>::new(network.clone());
+    let session_manager = Manager::new(commands_for_service, messages_for_service);
+    let network = Service::new(
+        network.clone(),
+        IO::new(messages_from_user, messages_for_user, commands_from_io),
+    );
 
+    //TODO: we should retry
+    let network_manager_task = async move {
+        connection_io
+            .run(connection_manager)
+            .await
+            .expect("Failed to run new network manager")
+    };
+    spawn_handle.spawn("aleph/network_manager", network_manager_task);
     let network_task = async move { network.run().await };
     spawn_handle.spawn("aleph/network", network_task);
 
     debug!(target: "afa", "Consensus network has started.");
 
+    let authorities = client
+        .runtime_api()
+        .authorities(&BlockId::Number(<NumberFor<B>>::saturated_from(0u32)))
+        .unwrap();
```

Comment:
The comment above is related, but also at least change this to `expect`.

Reply:
Done